App/Sagas/Migration.ts

```diff
@@ -32,6 +44,71 @@ export function * migrate(dispatch: Dispatch) {
   yield call(letScreenSleep)
 }
 
+// Can be run on each node online
+export function * runRecurringMigrationTasks () {
+  const announcement = yield select(getAnnouncement)
+  if (announcement) {
+    const {peerId, address, username, previous} = announcement
+    try {
+      yield call(announceMigration, peerId, address, username, previous)
+    } finally {
+      // If no error, mark as successful
+      yield put(MigrationActions.announceSuccess())
+    }
+  }
+  const peers = yield select(getNetwork)
+  for (const peer of peers) {
+    try {
+      // for each contact ask if they've migrated
+      const contact = yield call(findContact, peer)
+      if (contact) {
+        yield call(addContact, contact.peerId, contact.address, contact.username || '')
+      } else {
+        throw new Error('peer not found')
+      }
+    } finally {
+      // If no error, mark as successful
+      yield put(MigrationActions.connectionSuccess(peer))
```

Comment:
Second, just put this in the `if (contact)` block. We only want to consider it a success and remove that contact from redux if we've found the contact.